### ISSUE TITLE
unified_lock locked, waiting 1 second

### ISSUE THREAD

**Issue by 2232729885** [Project-HAMi/HAMi#976]
**What happened**: pytorch分布式训练，单机多卡训练报错，日志如下：

![Image](https://github.com/user-attachments/assets/6192a647-0e29-47e3-9949-3e618a793e2a)

![Image](https://github.com/user-attachments/assets/1443a8fe-f2aa-4077-8009-52bc0ea07a92)

**What you expected to happen**: pytorch分布式训练，单机多卡训练正常

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

- The output of `nvidia-smi -a` on your host 

![Image](https://github.com/user-attachments/assets/fa6abff0-6aba-4e4a-9e5b-595d73c10775)

- Your docker or containerd configuration file (e.g: `/etc/docker/daemon.json`)
- The hami-device-plugin container logs

![Image](https://github.com/user-attachments/assets/87509c6a-e558-4658-8733-c4bc47815393)

- The hami-scheduler container logs

![Image](https://github.com/user-attachments/assets/451b84c2-9c76-41e8-b57a-55ddbd29d183)

- The kubelet logs on the node (e.g: `sudo journalctl -r -u kubelet`)
- Any relevant kernel output lines from `dmesg`

**Environment**:
- HAMi version:  v2.4.1
- nvidia driver or other AI device driver version:
- Docker version from `docker version`
- Docker command, image and tag used
- Kernel version from `uname -a`
- Others:

**Comment by 2232729885** [comment]
补充 yaml:
`apiVersion: kubeflow.org/v1
kind: PyTorchJob
metadata:
  name: train-task-17171927680192
  namespace: algorithm-platform
spec:
  pytorchReplicaSpecs:
    Master:
      replicas: 1
      restartPolicy: Never
      template:
        metadata:
          annotations:
            sidecar.istio.io/inject: 'false'
        spec:
          containers:
            - command:
                - /bin/bash
                - '-c'
                - >-
                  mkdir -p /workspace/models && mkdir -p /root/bentoml && ln -s
                  /workspace/models /root/bentoml/models && cd /workspace &&
                  torchrun --nproc_per_node=4 multigpu_torchrun.py 10 2
              image: >-
                192.168.124.230:30002/algorithm-platform/pytorch/pytorch:2.1.0-cuda12.1-cudnn8-runtime
              imagePullPolicy: IfNotPresent
              name: pytorch
              resources:
                limits:
                  nvidia.com/gpu: '8'
                  nvidia.com/gpumem: '8192'
                requests: {}
              volumeMounts:
                - mountPath: /workspace
                  name: train-task-volume-17171927680192
                - mountPath: /dev/shm
                  name: dshm
          tolerations:
            - effect: NoSchedule
              key: nvidia.com/gpu
              operator: Exists
          volumes:
            - name: train-task-volume-17171927680192
              persistentVolumeClaim:
                claimName: train-task-pvc-17171927680192
                readOnly: false
            - emptyDir:
                medium: Memory
                sizeLimit: 2Gi
              name: dshm`

**Comment by 2232729885** [comment]
补充：
分配4张vgpu卡时正常，日志如下：
[2025-04-02 09:38:33,918] torch.distributed.run: [WARNING] *****************************************
[2025-04-02 09:38:33,918] torch.distributed.run: [WARNING] Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
[2025-04-02 09:38:33,918] torch.distributed.run: [WARNING] *****************************************
[HAMI-core Msg(12:139970323594432:libvgpu.c:836)]: Initializing.....
[HAMI-core Msg(13:140114944089280:libvgpu.c:836)]: Initializing.....
[HAMI-core Msg(14:140318898224320:libvgpu.c:836)]: Initializing.....
[HAMI-core Msg(11:140410923105472:libvgpu.c:836)]: Initializing.....
[HAMI-core Msg(12:139970323594432:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(11:140410923105472:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(14:140318898224320:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(13:140114944089280:libvgpu.c:855)]: Initialized
Loading snapshot
Resuming training from snapshot at Epoch 8
[HAMI-core Msg(11:140410923105472:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(14:140318898224320:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(12:139970323594432:libvgpu.c:855)]: Initialized
Loading snapshot
Resuming training from snapshot at Epoch 8
[HAMI-core Msg(14:140318898224320:utils.c:27)]: unified_lock locked, waiting 1 second...
[HAMI-core Msg(11:140410923105472:libvgpu.c:855)]: Initialized
Loading snapshot
Resuming training from snapshot at Epoch 8
[HAMI-core Msg(14:140318898224320:libvgpu.c:855)]: Initialized
Loading snapshot
Resuming training from snapshot at Epoch 8
[GPU1] Epoch 8 | Batchsize: 32 | Steps: 16
[GPU0] Epoch 8 | Batchsize: 32 | Steps: 16
[GPU3] Epoch 8 | Batchsize: 32 | Steps: 16
[GPU2] Epoch 8 | Batchsize: 32 | Steps: 16
[GPU1] Epoch 9 | Batchsize: 32 | Steps: 16[GPU3] Epoch 9 | Batchsize: 32 | Steps: 16

[GPU2] Epoch 9 | Batchsize: 32 | Steps: 16
Epoch 8 | Training snapshot saved at snapshot.pt
[GPU0] Epoch 9 | Batchsize: 32 | Steps: 16
[HAMI-core Msg(12:139970323594432:multiprocess_memory_limit.c:497)]: Calling exit handler 12
[HAMI-core Msg(11:140410923105472:multiprocess_memory_limit.c:497)]: Calling exit handler 11
[HAMI-core Msg(14:140318898224320:multiprocess_memory_limit.c:497)]: Calling exit handler 14
[HAMI-core Msg(13:140114944089280:multiprocess_memory_limit.c:497)]: Calling exit handler 13
[HAMI-core Msg(1:140193318890688:multiprocess_memory_limit.c:497)]: Calling exit handler 1

但是分配8张就报错，我的机器一共8张卡
